external-api/src/v1/controllers/estore/eStore.controller.ts

```diff
@@ -1,161 +1,237 @@
-        auditDetails: generateSystemAuditDetails(),
+    const cronJobExists = await cronJobLogs.findOne({ 'payload.dealId': { $eq: new ObjectId(eStoreData.dealId) } });
+
+    if (!cronJobExists) {
+      /**
+       * Send `201` status code back to avoid
+       * `TFM-API` awaiting.
+       */
+      console.info('Creating new CRON job for deal %s', eStoreData.dealIdentifier);
+      res.status(HttpStatusCode.Created).send({ status: HttpStatusCode.Created, message: 'eStore job accepted' });
+
+      // Step 1: Add CRON job to the collection
+      const { insertedId: _id } = await cronJobLogs.insertOne({
+        payload: eStoreData,
+        timestamp: getNowAsEpoch(),
+        cron: {
+          site: { status: ESTORE_CRON_STATUS.PENDING },
+          term: { status: ESTORE_CRON_STATUS.PENDING },
+          buyer: { status: ESTORE_CRON_STATUS.PENDING },
+          deal: { status: ESTORE_CRON_STATUS.PENDING },
+          facility: { status: ESTORE_CRON_STATUS.PENDING },
+        },
       });
 
-      console.info('API Call: Checking if the site exists');
-      const siteExistsResponse = await siteExists(eStoreData.exporterName);
-      // check if site exists in eStore
-      if (siteExistsResponse?.data?.status === ESTORE_SITE_STATUS.CREATED) {
-        // update the database to indicate that the site exists in eStore
-        await cronJobLogsCollection.updateOne(
-          { dealId: { $eq: eStoreData.dealId } },
-          { $set: { siteExists: true, siteId: siteExistsResponse.data.siteId, auditDetails: generateSystemAuditDetails() } },
+      // Step 2: Site exists check
+      console.info('Initiating eStore site existence check for exporter %s', eStoreData.exporterName);
+      const siteExistsResponse: SiteExistsResponse | EstoreErrorResponse = await siteExists(eStoreData.exporterName);
+
+      const created = siteExistsResponse?.data?.status === ESTORE_SITE_STATUS.CREATED;
+      const provisioning = siteExistsResponse?.data?.status === ESTORE_SITE_STATUS.PROVISIONING;
+      const absent = siteExistsResponse?.data?.status === HttpStatusCode.NotFound;
+
+      // Step 3: Site already exists in eStore
+      if (created) {
+        /**
+         * Update record-set with the site name.
+         * Update `cron-job-logs`
+         */
+        await cronJobLogs.updateOne(
+          { _id: { $eq: new ObjectId(_id) } },
+          {
+            $set: {
+              'response.site.siteId': siteExistsResponse.data.siteId,
+              'cron.site.create': {
+                status: ESTORE_CRON_STATUS.COMPLETED,
+                timestamp: getNowAsEpoch(),
+              },
+              'cron.site.status': ESTORE_CRON_STATUS.COMPLETED,
+            },
+          },
         );
 
-        eStoreData.siteId = siteExistsResponse.data.siteId;
+        // Update `tfm-deals`
+        await tfmDeals.updateOne({ _id: { $eq: new ObjectId(_id) } }, { $set: { 'tfm.estore.siteName': siteExistsResponse.data.siteId } });
```

Comment:
I think this `_id` should be the deal id, rather than `cron-job-logs` id

Reply:
Now that I am picking up `insetedId` I can use `_id`.

Reply:
The `_id` comes from the line
```
      const { insertedId: _id } = await cronJobLogs.insertOne({
```
But this line is updating a `tfm-deals` document which will have a different Id

Reply:
Apologies, have rectified.